ASTImport crash during visit of a ClassTemplateSpecializationDecl

```
ANALYZE (CTU loaded AST for source file): ./report_debug/ctu-dir/i386/ast/home/egbomrt/WORK/DebugCrashes/CodeChecker_log/reports/bns__2017_11_16__15_34_00/failed/Alarm.cc_7817343c04f25df6faf81ba12c6298da/sources-root/local/scratch/repos/ML66/ml66_nps/nps/bns_app_src/platform/eqm/lich/src/MibDb.cc
clang-5.0: ../../git/llvm/tools/clang/include/clang/AST/DeclContextInternals.h:116: void clang::StoredDeclsList::remove(clang::NamedDecl *): Assertion `I != Vec.end() && "list does not contain decl"' failed.
#0 0x00007fed7dfbb0b9 llvm::sys::PrintStackTrace(llvm::raw_ostream&) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/lib/Support/Unix/Signals.inc:398:11
#1 0x00007fed7dfbb269 PrintStackTraceSignalHandler(void*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/lib/Support/Unix/Signals.inc:462:1
#2 0x00007fed7dfb98c3 llvm::sys::RunSignalHandlers() /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/lib/Support/Signals.cpp:0:5
#3 0x00007fed7dfbb5c4 SignalHandler(int) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/lib/Support/Unix/Signals.inc:252:1
#4 0x00007fed7d22a390 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x11390)
#5 0x00007fed7a63d428 gsignal /build/glibc-bfm8X4/glibc-2.23/signal/../sysdeps/unix/sysv/linux/raise.c:54:0
#6 0x00007fed7a63f02a abort /build/glibc-bfm8X4/glibc-2.23/stdlib/abort.c:91:0
#7 0x00007fed7a635bd7 __assert_fail_base /build/glibc-bfm8X4/glibc-2.23/assert/assert.c:92:0
#8 0x00007fed7a635c82 (/lib/x86_64-linux-gnu/libc.so.6+0x2dc82)
#9 0x00007fed76e3987c clang::StoredDeclsList::remove(clang::NamedDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/include/clang/AST/DeclContextInternals.h:117:5
#10 0x00007fed76e3643c clang::DeclContext::removeDecl(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/DeclBase.cpp:1365:7
#11 0x00007fed76d3744e clang::ASTNodeImporter::ImportDeclContext(clang::DeclContext*, bool, clang::DeclContext*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:1010:5
#12 0x00007fed76d36dc0 clang::ASTNodeImporter::ImportDefinition(clang::RecordDecl*, clang::RecordDecl*, clang::ASTNodeImporter::ImportDefinitionKind) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:1132:3
#13 0x00007fed76d4b173 clang::ASTNodeImporter::VisitClassTemplateSpecializationDecl(clang::ClassTemplateSpecializationDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:4112:7
#14 0x00007fed76d7ff18 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::VisitClassTemplatePartialSpecializationDecl(clang::ClassTemplatePartialSpecializationDecl*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/DeclNodes.inc:259:1
#15 0x00007fed76d6a6d8 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/DeclNodes.inc:259:1
#16 0x00007fed76d3326a clang::ASTImporter::Import(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6422:9
#17 0x00007fed76d37377 clang::ASTNodeImporter::ImportDeclContext(clang::DeclContext*, bool, clang::DeclContext*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:1007:30
#18 0x00007fed76d39b36 clang::ASTNodeImporter::VisitNamespaceDecl(clang::NamespaceDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:1560:10
#19 0x00007fed76d6a4a8 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/DeclNodes.inc:105:1
#20 0x00007fed76d3326a clang::ASTImporter::Import(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6422:9
#21 0x00007fed76d355ee clang::ASTImporter::ImportContext(clang::DeclContext*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6458:23
#22 0x00007fed76d35412 clang::ASTNodeImporter::ImportDeclParts(clang::NamedDecl*, clang::DeclContext*&, clang::DeclContext*&, clang::DeclarationName&, clang::NamedDecl*&, clang::SourceLocation&) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:913:3
#23 0x00007fed76d49d1c clang::ASTNodeImporter::VisitClassTemplateDecl(clang::ClassTemplateDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:3899:7
#24 0x00007fed76d6a5dc clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/DeclNodes.inc:195:1
#25 0x00007fed76d3326a clang::ASTImporter::Import(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6422:9
#26 0x00007fed76d49c41 clang::ASTNodeImporter::VisitClassTemplateDecl(clang::ClassTemplateDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:3886:11
#27 0x00007fed76d6a5dc clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/DeclNodes.inc:195:1
#28 0x00007fed76d3326a clang::ASTImporter::Import(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6422:9
#29 0x00007fed76d37377 clang::ASTNodeImporter::ImportDeclContext(clang::DeclContext*, bool, clang::DeclContext*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:1007:30
#30 0x00007fed76d39b36 clang::ASTNodeImporter::VisitNamespaceDecl(clang::NamespaceDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:1560:10
#31 0x00007fed76d6a4a8 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/DeclNodes.inc:105:1
#32 0x00007fed76d3326a clang::ASTImporter::Import(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6422:9
#33 0x00007fed76d355ee clang::ASTImporter::ImportContext(clang::DeclContext*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6458:23
#34 0x00007fed76d35412 clang::ASTNodeImporter::ImportDeclParts(clang::NamedDecl*, clang::DeclContext*&, clang::DeclContext*&, clang::DeclarationName&, clang::NamedDecl*&, clang::SourceLocation&) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:913:3
#35 0x00007fed76d49d1c clang::ASTNodeImporter::VisitClassTemplateDecl(clang::ClassTemplateDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:3899:7
#36 0x00007fed76d6a5dc clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/DeclNodes.inc:195:1
#37 0x00007fed76d3326a clang::ASTImporter::Import(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6422:9
#38 0x00007fed76d34290 clang::ASTImporter::Import(clang::TemplateName) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6664:19
#39 0x00007fed76d34008 clang::ASTNodeImporter::VisitTemplateSpecializationType(clang::TemplateSpecializationType const*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:792:38
#40 0x00007fed76d69e2c clang::TypeVisitor<clang::ASTNodeImporter, clang::QualType>::Visit(clang::Type const*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/include/clang/AST/TypeNodes.def:98:1
#41 0x00007fed76d31556 clang::ASTImporter::Import(clang::QualType) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6372:27
#42 0x00007fed76d34ba7 clang::ASTImporter::Import(clang::NestedNameSpecifier*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6575:20
#43 0x00007fed76d34c8a clang::ASTNodeImporter::VisitDependentNameType(clang::DependentNameType const*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:831:24
#44 0x00007fed76d69e9c clang::TypeVisitor<clang::ASTNodeImporter, clang::QualType>::Visit(clang::Type const*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/include/clang/AST/TypeNodes.def:103:1
#45 0x00007fed76d31556 clang::ASTImporter::Import(clang::QualType) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6372:27
#46 0x00007fed76d3245c clang::ASTNodeImporter::VisitLValueReferenceType(clang::LValueReferenceType const*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:453:37
#47 0x00007fed76d69b1c clang::TypeVisitor<clang::ASTNodeImporter, clang::QualType>::Visit(clang::Type const*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/include/clang/AST/TypeNodes.def:67:1
#48 0x00007fed76d31556 clang::ASTImporter::Import(clang::QualType) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6372:27
#49 0x00007fed76d32e62 clang::ASTNodeImporter::VisitFunctionProtoType(clang::FunctionProtoType const*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:578:33
#50 0x00007fed76d69c34 clang::TypeVisitor<clang::ASTNodeImporter, clang::QualType>::Visit(clang::Type const*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/include/clang/AST/TypeNodes.def:79:1
#51 0x00007fed76d31556 clang::ASTImporter::Import(clang::QualType) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6372:27
#52 0x00007fed76d3e72f clang::ASTNodeImporter::VisitFunctionDecl(clang::FunctionDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:2214:25
#53 0x00007fed76d6a87c clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/DeclNodes.inc:389:1
#54 0x00007fed76d3326a clang::ASTImporter::Import(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6422:9
#55 0x00007fed76d4c605 clang::ASTNodeImporter::VisitFunctionTemplateDecl(clang::FunctionTemplateDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:4333:7
#56 0x00007fed76d6a5f8 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/DeclNodes.inc:201:1
#57 0x00007fed76d3326a clang::ASTImporter::Import(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6422:9
#58 0x00007fed76d37377 clang::ASTNodeImporter::ImportDeclContext(clang::DeclContext*, bool, clang::DeclContext*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:1007:30
#59 0x00007fed76d39b36 clang::ASTNodeImporter::VisitNamespaceDecl(clang::NamespaceDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:1560:10
#60 0x00007fed76d6a4a8 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/DeclNodes.inc:105:1
#61 0x00007fed76d3326a clang::ASTImporter::Import(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6422:9
#62 0x00007fed76d355ee clang::ASTImporter::ImportContext(clang::DeclContext*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6458:23
#63 0x00007fed76d35412 clang::ASTNodeImporter::ImportDeclParts(clang::NamedDecl*, clang::DeclContext*&, clang::DeclContext*&, clang::DeclarationName&, clang::NamedDecl*&, clang::SourceLocation&) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:913:3
#64 0x00007fed76d4c56c clang::ASTNodeImporter::VisitFunctionTemplateDecl(clang::FunctionTemplateDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:4326:7
#65 0x00007fed76d6a5f8 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/DeclNodes.inc:201:1
#66 0x00007fed76d3326a clang::ASTImporter::Import(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6422:9
#67 0x00007fed76d58c2a clang::ASTNodeImporter::VisitUnresolvedLookupExpr(clang::UnresolvedLookupExpr*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6061:26
#68 0x00007fed76d6c61f clang::StmtVisitorBase<clang::make_ptr, clang::ASTNodeImporter, clang::Stmt*>::Visit(clang::Stmt*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/StmtNodes.inc:721:1
#69 0x00007fed76d3b593 clang::ASTImporter::Import(clang::Stmt*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6527:9
#70 0x00007fed76d32965 clang::ASTImporter::Import(clang::Expr*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6513:10
#71 0x00007fed76d59ca1 clang::ASTNodeImporter::VisitCallExpr(clang::CallExpr*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6176:9
#72 0x00007fed76d6bfab clang::StmtVisitorBase<clang::make_ptr, clang::ASTNodeImporter, clang::Stmt*>::Visit(clang::Stmt*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/StmtNodes.inc:329:1
#73 0x00007fed76d3b593 clang::ASTImporter::Import(clang::Stmt*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6527:9
#74 0x00007fed76d32965 clang::ASTImporter::Import(clang::Expr*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6513:10
#75 0x00007fed76d7c1a7 bool clang::ASTNodeImporter::ImportArrayChecked<clang::Expr* const*, clang::Expr**>(clang::Expr* const*, clang::Expr* const*, clang::Expr**)::{lambda(clang::Expr*)#1}::operator()(clang::Expr*) const /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:325:36
#76 0x00007fed76d7c14a clang::Expr** std::transform<clang::Expr* const*, clang::Expr**, bool clang::ASTNodeImporter::ImportArrayChecked<clang::Expr* const*, clang::Expr**>(clang::Expr* const*, clang::Expr* const*, clang::Expr**)::{lambda(clang::Expr*)#1}>(clang::Expr* const*, clang::Expr* const*, clang::Expr**, bool clang::ASTNodeImporter::ImportArrayChecked<clang::Expr* const*, clang::Expr**>(clang::Expr* const*, clang::Expr* const*, clang::Expr**)::{lambda(clang::Expr*)#1}) /usr/bin/../lib/gcc/x86_64-linux-gnu/7.2.0/../../../../include/c++/7.2.0/bits/stl_algo.h:4306:3
#77 0x00007fed76d7c0b0 bool clang::ASTNodeImporter::ImportArrayChecked<clang::Expr* const*, clang::Expr**>(clang::Expr* const*, clang::Expr* const*, clang::Expr**) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:331:14
#78 0x00007fed76d6479c bool clang::ASTNodeImporter::ImportContainerChecked<llvm::ArrayRef<clang::Expr*>, llvm::SmallVector<clang::Expr*, 4u> >(llvm::ArrayRef<clang::Expr*> const&, llvm::SmallVector<clang::Expr*, 4u>&) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:337:7
#79 0x00007fed76d523d6 clang::ASTNodeImporter::VisitParenListExpr(clang::ParenListExpr*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:5194:7
#80 0x00007fed76d6c68f clang::StmtVisitorBase<clang::make_ptr, clang::ASTNodeImporter, clang::Stmt*>::Visit(clang::Stmt*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/StmtNodes.inc:749:1
#81 0x00007fed76d3b593 clang::ASTImporter::Import(clang::Stmt*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6527:9
#82 0x00007fed76d32965 clang::ASTImporter::Import(clang::Expr*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6513:10
#83 0x00007fed76d3f846 clang::ASTImporter::Import(clang::CXXCtorInitializer*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6820:9
#84 0x00007fed76d3eae4 clang::ASTNodeImporter::VisitFunctionDecl(clang::FunctionDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:2245:13
#85 0x00007fed76d3fefd clang::ASTNodeImporter::VisitCXXMethodDecl(clang::CXXMethodDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:2347:3
#86 0x00007fed76d3ff2d clang::ASTNodeImporter::VisitCXXConstructorDecl(clang::CXXConstructorDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:2351:3
#87 0x00007fed76d6a8d0 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/DeclNodes.inc:403:1
#88 0x00007fed76d3326a clang::ASTImporter::Import(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6422:9
#89 0x00007fed76d4c605 clang::ASTNodeImporter::VisitFunctionTemplateDecl(clang::FunctionTemplateDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:4333:7
#90 0x00007fed76d6a5f8 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/DeclNodes.inc:201:1
#91 0x00007fed76d3326a clang::ASTImporter::Import(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6422:9
#92 0x00007fed76d37377 clang::ASTNodeImporter::ImportDeclContext(clang::DeclContext*, bool, clang::DeclContext*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:1007:30
#93 0x00007fed76d36dc0 clang::ASTNodeImporter::ImportDefinition(clang::RecordDecl*, clang::RecordDecl*, clang::ASTNodeImporter::ImportDefinitionKind) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:1132:3
#94 0x00007fed76d4b173 clang::ASTNodeImporter::VisitClassTemplateSpecializationDecl(clang::ClassTemplateSpecializationDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:4112:7
#95 0x00007fed76d7ff18 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::VisitClassTemplatePartialSpecializationDecl(clang::ClassTemplatePartialSpecializationDecl*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/DeclNodes.inc:259:1
#96 0x00007fed76d6a6d8 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/DeclNodes.inc:259:1
#97 0x00007fed76d3326a clang::ASTImporter::Import(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6422:9
#98 0x00007fed76d37377 clang::ASTNodeImporter::ImportDeclContext(clang::DeclContext*, bool, clang::DeclContext*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:1007:30
#99 0x00007fed76d39b36 clang::ASTNodeImporter::VisitNamespaceDecl(clang::NamespaceDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:1560:10
#100 0x00007fed76d6a4a8 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/DeclNodes.inc:105:1
#101 0x00007fed76d3326a clang::ASTImporter::Import(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6422:9
#102 0x00007fed76d355ee clang::ASTImporter::ImportContext(clang::DeclContext*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6458:23
#103 0x00007fed76d35412 clang::ASTNodeImporter::ImportDeclParts(clang::NamedDecl*, clang::DeclContext*&, clang::DeclContext*&, clang::DeclarationName&, clang::NamedDecl*&, clang::SourceLocation&) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:913:3
#104 0x00007fed76d49d1c clang::ASTNodeImporter::VisitClassTemplateDecl(clang::ClassTemplateDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:3899:7
#105 0x00007fed76d6a5dc clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/DeclNodes.inc:195:1
#106 0x00007fed76d3326a clang::ASTImporter::Import(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6422:9
#107 0x00007fed76d49c41 clang::ASTNodeImporter::VisitClassTemplateDecl(clang::ClassTemplateDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:3886:11
#108 0x00007fed76d6a5dc clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/DeclNodes.inc:195:1
#109 0x00007fed76d3326a clang::ASTImporter::Import(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6422:9
#110 0x00007fed76d37377 clang::ASTNodeImporter::ImportDeclContext(clang::DeclContext*, bool, clang::DeclContext*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:1007:30
#111 0x00007fed76d39b36 clang::ASTNodeImporter::VisitNamespaceDecl(clang::NamespaceDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:1560:10
#112 0x00007fed76d6a4a8 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/DeclNodes.inc:105:1
#113 0x00007fed76d3326a clang::ASTImporter::Import(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6422:9
#114 0x00007fed76d355ee clang::ASTImporter::ImportContext(clang::DeclContext*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6458:23
#115 0x00007fed76d35412 clang::ASTNodeImporter::ImportDeclParts(clang::NamedDecl*, clang::DeclContext*&, clang::DeclContext*&, clang::DeclarationName&, clang::NamedDecl*&, clang::SourceLocation&) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:913:3
#116 0x00007fed76d4c56c clang::ASTNodeImporter::VisitFunctionTemplateDecl(clang::FunctionTemplateDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:4326:7
#117 0x00007fed76d6a5f8 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/DeclNodes.inc:201:1
#118 0x00007fed76d3326a clang::ASTImporter::Import(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6422:9
#119 0x00007fed76d58c2a clang::ASTNodeImporter::VisitUnresolvedLookupExpr(clang::UnresolvedLookupExpr*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6061:26
#120 0x00007fed76d6c61f clang::StmtVisitorBase<clang::make_ptr, clang::ASTNodeImporter, clang::Stmt*>::Visit(clang::Stmt*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/StmtNodes.inc:721:1
#121 0x00007fed76d3b593 clang::ASTImporter::Import(clang::Stmt*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6527:9
#122 0x00007fed76d32965 clang::ASTImporter::Import(clang::Expr*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6513:10
#123 0x00007fed76d59ca1 clang::ASTNodeImporter::VisitCallExpr(clang::CallExpr*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6176:9
#124 0x00007fed76d81808 clang::StmtVisitorBase<clang::make_ptr, clang::ASTNodeImporter, clang::Stmt*>::VisitCXXOperatorCallExpr(clang::CXXOperatorCallExpr*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/StmtNodes.inc:345:1
#125 0x00007fed76d6bfff clang::StmtVisitorBase<clang::make_ptr, clang::ASTNodeImporter, clang::Stmt*>::Visit(clang::Stmt*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/StmtNodes.inc:345:1
#126 0x00007fed76d3b593 clang::ASTImporter::Import(clang::Stmt*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6527:9
#127 0x00007fed76d32965 clang::ASTImporter::Import(clang::Expr*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6513:10
#128 0x00007fed76d4f190 clang::ASTNodeImporter::VisitReturnStmt(clang::ReturnStmt*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:4747:9
#129 0x00007fed76d6cec3 clang::StmtVisitorBase<clang::make_ptr, clang::ASTNodeImporter, clang::Stmt*>::Visit(clang::Stmt*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/StmtNodes.inc:1219:1
#130 0x00007fed76d3b593 clang::ASTImporter::Import(clang::Stmt*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6527:9
#131 0x00007fed76d7b057 bool clang::ASTNodeImporter::ImportArrayChecked<clang::Stmt**, clang::Stmt**>(clang::Stmt**, clang::Stmt**, clang::Stmt**)::{lambda(clang::Stmt*)#1}::operator()(clang::Stmt*) const /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:325:36
#132 0x00007fed76d7affa clang::Stmt** std::transform<clang::Stmt**, clang::Stmt**, bool clang::ASTNodeImporter::ImportArrayChecked<clang::Stmt**, clang::Stmt**>(clang::Stmt**, clang::Stmt**, clang::Stmt**)::{lambda(clang::Stmt*)#1}>(clang::Stmt**, clang::Stmt**, clang::Stmt**, bool clang::ASTNodeImporter::ImportArrayChecked<clang::Stmt**, clang::Stmt**>(clang::Stmt**, clang::Stmt**, clang::Stmt**)::{lambda(clang::Stmt*)#1}) /usr/bin/../lib/gcc/x86_64-linux-gnu/7.2.0/../../../../include/c++/7.2.0/bits/stl_algo.h:4306:3
#133 0x00007fed76d7af60 bool clang::ASTNodeImporter::ImportArrayChecked<clang::Stmt**, clang::Stmt**>(clang::Stmt**, clang::Stmt**, clang::Stmt**) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:331:14
#134 0x00007fed76d618ac bool clang::ASTNodeImporter::ImportContainerChecked<llvm::iterator_range<clang::Stmt**>, llvm::SmallVector<clang::Stmt*, 8u> >(llvm::iterator_range<clang::Stmt**> const&, llvm::SmallVector<clang::Stmt*, 8u>&) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:337:7
#135 0x00007fed76d4d86c clang::ASTNodeImporter::VisitCompoundStmt(clang::CompoundStmt*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:4505:7
#136 0x00007fed76d6bb32 clang::StmtVisitorBase<clang::make_ptr, clang::ASTNodeImporter, clang::Stmt*>::Visit(clang::Stmt*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/StmtNodes.inc:79:1
#137 0x00007fed76d3b593 clang::ASTImporter::Import(clang::Stmt*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6527:9
#138 0x00007fed76d3f6c5 clang::ASTNodeImporter::VisitFunctionDecl(clang::FunctionDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:2330:15
#139 0x00007fed76d6a87c clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/DeclNodes.inc:389:1
#140 0x00007fed76d3326a clang::ASTImporter::Import(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6422:9
#141 0x00007fed76d4c605 clang::ASTNodeImporter::VisitFunctionTemplateDecl(clang::FunctionTemplateDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:4333:7
#142 0x00007fed76d6a5f8 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/DeclNodes.inc:201:1
#143 0x00007fed76d3326a clang::ASTImporter::Import(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6422:9
#144 0x00007fed76d37377 clang::ASTNodeImporter::ImportDeclContext(clang::DeclContext*, bool, clang::DeclContext*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:1007:30
#145 0x00007fed76d39b36 clang::ASTNodeImporter::VisitNamespaceDecl(clang::NamespaceDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:1560:10
#146 0x00007fed76d6a4a8 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/DeclNodes.inc:105:1
#147 0x00007fed76d3326a clang::ASTImporter::Import(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6422:9
#148 0x00007fed76d355ee clang::ASTImporter::ImportContext(clang::DeclContext*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6458:23
#149 0x00007fed76d35412 clang::ASTNodeImporter::ImportDeclParts(clang::NamedDecl*, clang::DeclContext*&, clang::DeclContext*&, clang::DeclarationName&, clang::NamedDecl*&, clang::SourceLocation&) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:913:3
#150 0x00007fed76d49d1c clang::ASTNodeImporter::VisitClassTemplateDecl(clang::ClassTemplateDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:3899:7
#151 0x00007fed76d6a5dc clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/DeclNodes.inc:195:1
#152 0x00007fed76d3326a clang::ASTImporter::Import(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6422:9
#153 0x00007fed76d34290 clang::ASTImporter::Import(clang::TemplateName) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6664:19
#154 0x00007fed76d34008 clang::ASTNodeImporter::VisitTemplateSpecializationType(clang::TemplateSpecializationType const*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:792:38
#155 0x00007fed76d69e2c clang::TypeVisitor<clang::ASTNodeImporter, clang::QualType>::Visit(clang::Type const*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/include/clang/AST/TypeNodes.def:98:1
#156 0x00007fed76d31556 clang::ASTImporter::Import(clang::QualType) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6372:27
#157 0x00007fed76d34930 clang::ASTNodeImporter::VisitElaboratedType(clang::ElaboratedType const*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:822:35
#158 0x00007fed76d69da0 clang::TypeVisitor<clang::ASTNodeImporter, clang::QualType>::Visit(clang::Type const*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/include/clang/AST/TypeNodes.def:93:1
#159 0x00007fed76d31556 clang::ASTImporter::Import(clang::QualType) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6372:27
#160 0x00007fed76d4041b clang::ASTNodeImporter::VisitFieldDecl(clang::FieldDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:2414:25
#161 0x00007fed76d6a828 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/DeclNodes.inc:369:1
#162 0x00007fed76d3326a clang::ASTImporter::Import(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6422:9
#163 0x00007fed76d37377 clang::ASTNodeImporter::ImportDeclContext(clang::DeclContext*, bool, clang::DeclContext*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:1007:30
#164 0x00007fed76d36dc0 clang::ASTNodeImporter::ImportDefinition(clang::RecordDecl*, clang::RecordDecl*, clang::ASTNodeImporter::ImportDefinitionKind) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:1132:3
#165 0x00007fed76d3d4fc clang::ASTNodeImporter::VisitRecordDecl(clang::RecordDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:2057:7
#166 0x00007fed76d7fee8 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::VisitCXXRecordDecl(clang::CXXRecordDecl*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/DeclNodes.inc:251:1
#167 0x00007fed76d6a6a0 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/DeclNodes.inc:251:1
#168 0x00007fed76d3326a clang::ASTImporter::Import(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6422:9
#169 0x00007fed76d355ee clang::ASTImporter::ImportContext(clang::DeclContext*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6458:23
#170 0x00007fed76d35412 clang::ASTNodeImporter::ImportDeclParts(clang::NamedDecl*, clang::DeclContext*&, clang::DeclContext*&, clang::DeclarationName&, clang::NamedDecl*&, clang::SourceLocation&) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:913:3
#171 0x00007fed76d3ddbc clang::ASTNodeImporter::VisitFunctionDecl(clang::FunctionDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:2128:7
#172 0x00007fed76d3fefd clang::ASTNodeImporter::VisitCXXMethodDecl(clang::CXXMethodDecl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:2347:3
#173 0x00007fed76d6a8b4 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/tools/clang/include/clang/AST/DeclNodes.inc:399:1
#174 0x00007fed76d3326a clang::ASTImporter::Import(clang::Decl*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/AST/ASTImporter.cpp:6422:9
#175 0x00007fed6ffaa837 clang::tooling::CrossTranslationUnit::getCrossTUDefinition(clang::FunctionDecl const*, llvm::StringRef, llvm::StringRef, llvm::StringRef, bool) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/Tooling/CrossTranslationUnit.cpp:204:20
#176 0x00007fed7041202f clang::ento::AnyFunctionCall::getRuntimeDefinition() const /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/StaticAnalyzer/Core/CallEvent.cpp:384:37
#177 0x00007fed704130ba clang::ento::CXXInstanceCall::getRuntimeDefinition() const /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/StaticAnalyzer/Core/CallEvent.cpp:523:29
#178 0x00007fed7041362c clang::ento::CXXMemberCall::getRuntimeDefinition() const /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/StaticAnalyzer/Core/CallEvent.cpp:622:27
#179 0x00007fed704aa1f6 clang::ento::ExprEngine::defaultEvalCall(clang::ento::NodeBuilder&, clang::ento::ExplodedNode*, clang::ento::CallEvent const&) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/StaticAnalyzer/Core/ExprEngineCallAndReturn.cpp:915:34
#180 0x00007fed7042bd6f clang::ento::CheckerManager::runCheckersForEvalCall(clang::ento::ExplodedNodeSet&, clang::ento::ExplodedNodeSet const&, clang::ento::CallEvent const&, clang::ento::ExprEngine&) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/StaticAnalyzer/Core/CheckerManager.cpp:619:11
#181 0x00007fed704a9029 clang::ento::ExprEngine::evalCall(clang::ento::ExplodedNodeSet&, clang::ento::ExplodedNode*, clang::ento::CallEvent const&) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/StaticAnalyzer/Core/ExprEngineCallAndReturn.cpp:521:3
#182 0x00007fed704a8eeb clang::ento::ExprEngine::VisitCallExpr(clang::CallExpr const*, clang::ento::ExplodedNode*, clang::ento::ExplodedNodeSet&) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/StaticAnalyzer/Core/ExprEngineCallAndReturn.cpp:0:5
#183 0x00007fed70469764 clang::ento::ExprEngine::Visit(clang::Stmt const*, clang::ento::ExplodedNode*, clang::ento::ExplodedNodeSet&) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/StaticAnalyzer/Core/ExprEngine.cpp:1232:7
#184 0x00007fed704668e4 clang::ento::ExprEngine::ProcessStmt(clang::CFGStmt, clang::ento::ExplodedNode*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/StaticAnalyzer/Core/ExprEngine.cpp:547:5
#185 0x00007fed704665aa clang::ento::ExprEngine::processCFGElement(clang::CFGElement, clang::ento::ExplodedNode*, unsigned int, clang::ento::NodeBuilderContext*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/StaticAnalyzer/Core/ExprEngine.cpp:394:7
#186 0x00007fed7044ad66 clang::ento::CoreEngine::HandlePostStmt(clang::CFGBlock const*, unsigned int, clang::ento::ExplodedNode*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/StaticAnalyzer/Core/CoreEngine.cpp:532:1
#187 0x00007fed7044a663 clang::ento::CoreEngine::dispatchWorkItem(clang::ento::ExplodedNode*, clang::ProgramPoint, clang::ento::WorkListUnit const&) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/StaticAnalyzer/Core/CoreEngine.cpp:281:1
#188 0x00007fed7044a1f8 clang::ento::CoreEngine::ExecuteWorkList(clang::LocationContext const*, unsigned int, llvm::IntrusiveRefCntPtr<clang::ento::ProgramState const>) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/StaticAnalyzer/Core/CoreEngine.cpp:216:3
#189 0x00007fed71f69f45 clang::ento::ExprEngine::ExecuteWorkList(clang::LocationContext const*, unsigned int) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/include/clang/StaticAnalyzer/Core/PathSensitive/ExprEngine.h:114:19
#190 0x00007fed71f0ce6b (anonymous namespace)::AnalysisConsumer::ActionExprEngine(clang::Decl*, bool, clang::ento::ExprEngine::InliningModes, llvm::DenseSet<clang::Decl const*, llvm::DenseMapInfo<clang::Decl const*> >*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/StaticAnalyzer/Frontend/AnalysisConsumer.cpp:741:7
#191 0x00007fed71f0cc01 (anonymous namespace)::AnalysisConsumer::RunPathSensitiveChecks(clang::Decl*, clang::ento::ExprEngine::InliningModes, llvm::DenseSet<clang::Decl const*, llvm::DenseMapInfo<clang::Decl const*> >*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/StaticAnalyzer/Frontend/AnalysisConsumer.cpp:762:5
#192 0x00007fed71f0c8b5 (anonymous namespace)::AnalysisConsumer::HandleCode(clang::Decl*, unsigned int, clang::ento::ExprEngine::InliningModes, llvm::DenseSet<clang::Decl const*, llvm::DenseMapInfo<clang::Decl const*> >*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/StaticAnalyzer/Frontend/AnalysisConsumer.cpp:709:15
#193 0x00007fed71f0001a (anonymous namespace)::AnalysisConsumer::HandleDeclsCallGraph(unsigned int) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/StaticAnalyzer/Frontend/AnalysisConsumer.cpp:521:5
#194 0x00007fed71efe27d (anonymous namespace)::AnalysisConsumer::HandleTranslationUnit(clang::ASTContext&) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/StaticAnalyzer/Frontend/AnalysisConsumer.cpp:576:5
#195 0x00007fed7b78188e clang::MultiplexConsumer::HandleTranslationUnit(clang::ASTContext&) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/Frontend/MultiplexConsumer.cpp:0:15
#196 0x00007fed75000796 clang::ParseAST(clang::Sema&, bool, bool) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/Parse/ParseAST.cpp:159:13
#197 0x00007fed7b74b4ec clang::ASTFrontendAction::ExecuteAction() /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/Frontend/FrontendAction.cpp:1005:1
#198 0x00007fed7b74af60 clang::FrontendAction::Execute() /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/Frontend/FrontendAction.cpp:906:7
#199 0x00007fed7b6c40ba clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/Frontend/CompilerInstance.cpp:981:7
#200 0x00007fed7b28ecdf clang::ExecuteCompilerInvocation(clang::CompilerInstance*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/lib/FrontendTool/ExecuteCompilerInvocation.cpp:251:8
#201 0x0000000000426659 cc1_main(llvm::ArrayRef<char const*>, char const*, void*) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/tools/driver/cc1_main.cpp:221:11
#202 0x00000000004194c9 ExecuteCC1Tool(llvm::ArrayRef<char const*>, llvm::StringRef) /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/tools/driver/driver.cpp:306:5
#203 0x0000000000418100 main /home/egbomrt/WORK/llvm/build/debug/../../git/llvm/tools/clang/tools/driver/driver.cpp:387:5
#204 0x00007fed7a628830 __libc_start_main /build/glibc-bfm8X4/glibc-2.23/csu/../csu/libc-start.c:325:0
#205 0x00000000004170e9 _start (/home/egbomrt/WORK/llvm/build/debug/bin/clang-5.0+0x4170e9)
Stack dump:
0.      Program arguments: /home/egbomrt/WORK/llvm/build/debug/bin/clang-5.0 -cc1 -triple powerpc-montavista-linux-gnu -analyze -disable-free -main-file-name Alarm.cc -analyzer-store=region -analyzer-opt-analyze-nested-blocks -analyzer-eagerly-assume -analyzer-output plist -w -mrelocation-model static -mthread-model posix -fmath-errno -masm-verbose -mconstructor-aliases -target-cpu ppc -mfloat-abi hard -momit-leaf-frame-pointer -dwarf-column-info -debugger-tuning=gdb -resource-dir ./sources-root/proj/tn/tools/code_checker/CodeChecker_6.2.0/include -isystem ./sources-root/proj/tn/tools/code_checker/CodeChecker_6.2.0/include -isystem ./sources-root/proj/tn/build/toolchain/e5500/R1A03/montavista/tools/powerpc64-gnu/powerpc64-montavista-linux-gnu/include/c++/4.7.0 -isystem ./sources-root/proj/tn/build/toolchain/e5500/R1A03/montavista/tools/powerpc64-gnu/powerpc64-montavista-linux-gnu/include/c++/4.7.0/powerpc64-montavista-linux-gnu/32 -isystem ./sources-root/proj/tn/build/toolchain/e5500/R1A03/montavista/tools/powerpc64-gnu/powerpc64-montavista-linux-gnu/include/c++/4.7.0/backward -isystem ./sources-root/proj/tn/build/toolchain/e5500/R1A03/montavista/tools/powerpc64-gnu/lib/gcc/powerpc64-montavista-linux-gnu/4.7.0/include -isystem ./sources-root/proj/tn/build/toolchain/e5500/R1A03/montavista/tools/powerpc64-gnu/lib/gcc/powerpc64-montavista-linux-gnu/4.7.0/include-fixed -isystem ./sources-root/proj/tn/build/toolchain/e5500/R1A03/montavista/tools/powerpc64-gnu/powerpc64-montavista-linux-gnu/include -isystem ./sources-root/proj/tn/build/sysroot/e5500/R1A04/usr/include -D ARCH_e5500 -D BNS_APP_SECURITY_FEATURE -D PRODTYPE_NPU -I ./sources-root/local/scratch/repos/ML66/ml66_nps/nps/bns_app_src/include -I ./sources-root/local/scratch/repos/ML66/ml66_nps/nps/bns_app_src/include/lch -I ./sources-root/local/scratch/repos/ML66/ml66_nps/nps/bns_app_src/include/rph -I ./sources-root/local/scratch/repos/ML66/ml66_nps/nps/bns_app_src/include/ecps -I ./sources-root/local/scratch/repos/ML66/ml66_nps/nps/bns_app_src/include/tacacs -I ./sources-root/local/scratch/repos/ML66/ml66_nps/submodules/ml66_common -I ./sources-root/local/scratch/repos/ML66/ml66_nps/submodules/ml66_common/include -I ./sources-root/local/scratch/repos/ML66/ml66_nps/submodules/ml66_common/include/envoy/target/h -I ./sources-root/local/scratch/repos/ML66/ml66_nps/submodules/ml66_common/include/xf -I ./sources-root/local/scratch/repos/ML66/ml66_nps/submodules/ml66_common/include/cmo -I ./sources-root/local/scratch/repos/ML66/ml66_nps/submodules/ml66_common/include/smo -I ./sources-root/local/scratch/repos/ML66/ml66_nps/submodules/ml66_common/include/tm -I ./sources-root/local/scratch/repos/ML66/ml66_nps/submodules/ml66_common/include/tnu -I ./sources-root/local/scratch/repos/ML66/ml66_nps/submodules/ml66_common/include/radius -I ./sources-root/local/scratch/repos/ML66/ml66_nps/submodules/ml66_common/include/ecps -I ./sources-root/local/scratch/repos/ML66/ml66_nps/submodules/ml66_common/include/eqm -I ./sources-root/local/scratch/repos/ML66/ml66_nps/submodules/ml66_common/src/platform/eqm/lich-lib/include -I ./sources-root/local/scratch/repos/ML66/ml66_nps/nps/bns_app_src/platform/eqm/lich/include -D RELEASE -isysroot ./sources-root/proj/tn/build/sysroot/e5500/R1A04 -internal-isystem ./sources-root/proj/tn/build/sysroot/e5500/R1A04/usr/local/include -internal-isystem ./sources-root/proj/tn/tools/code_checker/CodeChecker_6.2.0/include/include -internal-externc-isystem ./sources-root/proj/tn/build/sysroot/e5500/R1A04/include -internal-externc-isystem ./sources-root/proj/tn/build/sysroot/e5500/R1A04/usr/include -O2 -std=c++11 -fdeprecated-macro -fdebug-compilation-dir /home/egbomrt/WORK/DebugCrashes/CodeChecker_log/reports/bns__2017_11_16__15_34_00/failed/Alarm.cc_7817343c04f25df6faf81ba12c6298da -ferror-limit 19 -fmessage-length 115 -fno-signed-char -fobjc-runtime=gcc -fcxx-exceptions -fexceptions -fdiagnostics-show-option -fcolor-diagnostics -vectorize-loops -vectorize-slp -plugin checkercfg -load /home/egbomrt/WORK/codechecker_core_ws/build/debug/libericsson-checkers.so -analyzer-opt-analyze-headers -analyzer-output=plist-multi-file -analyzer-disable-checker alpha.clone.CloneChecker -analyzer-disable-checker alpha.core.BoolAssignment -analyzer-disable-checker alpha.core.CallAndMessageUnInitRefArg -analyzer-disable-checker alpha.core.CastSize -analyzer-disable-checker alpha.core.CastToStruct -analyzer-disable-checker alpha.core.Conversion -analyzer-disable-checker alpha.core.DynamicTypeChecker -analyzer-disable-checker alpha.core.FixedAddr -analyzer-disable-checker alpha.core.IdenticalExpr -analyzer-disable-checker alpha.core.PointerArithm -analyzer-disable-checker alpha.core.PointerSub -analyzer-disable-checker alpha.core.SizeofPtr -analyzer-disable-checker alpha.core.TestAfterDivZero -analyzer-disable-checker alpha.cplusplus.IteratorRange -analyzer-disable-checker alpha.cplusplus.MisusedMovedObject -analyzer-disable-checker alpha.deadcode.UnreachableCode -analyzer-disable-checker alpha.ericsson.cpp.BitwiseShift -analyzer-disable-checker alpha.ericsson.cpp.CtorUninitializedMember -analyzer-disable-checker alpha.ericsson.cpp.DirtyScalar -analyzer-disable-checker alpha.ericsson.cpp.ExceptionMisuse -analyzer-disable-checker alpha.ericsson.cpp.InvariablePtrBranch -analyzer-disable-checker alpha.ericsson.cpp.IteratorOutOfRange -analyzer-disable-checker alpha.ericsson.cpp.MismatchedIterator -analyzer-disable-checker alpha.ericsson.cpp.MissingHeaderGuard -analyzer-disable-checker alpha.ericsson.cpp.OStreamFormat -analyzer-disable-checker alpha.ericsson.cpp.PointerArithmetic -analyzer-disable-checker alpha.ericsson.cpp.Slicing -analyzer-disable-checker alpha.ericsson.cpp.stl.MissingIteratorAdapter -analyzer-disable-checker alpha.ericsson.cpp.style.PreferInit -analyzer-disable-checker alpha.ericsson.statisticsbased.SpecialReturnValue -analyzer-disable-checker alpha.ericsson.statisticsbased.UncheckedReturnValue -analyzer-disable-checker alpha.ericsson.statisticscollector.ReturnValueCheck -analyzer-disable-checker alpha.ericsson.statisticscollector.SpecialReturnValue -analyzer-disable-checker alpha.osx.cocoa.DirectIvarAssignment -analyzer-disable-checker alpha.osx.cocoa.DirectIvarAssignmentForAnnotatedFunctions -analyzer-disable-checker alpha.osx.cocoa.InstanceVariableInvalidation -analyzer-disable-checker alpha.osx.cocoa.MissingInvalidationMethod -analyzer-disable-checker alpha.osx.cocoa.localizability.PluralMisuseChecker -analyzer-disable-checker alpha.security.ArrayBound -analyzer-disable-checker alpha.security.ArrayBoundV2 -analyzer-disable-checker alpha.security.MallocOverflow -analyzer-disable-checker alpha.security.ReturnPtrRange -analyzer-disable-checker alpha.security.taint.TaintPropagation -analyzer-disable-checker alpha.unix.BlockInCriticalSection -analyzer-disable-checker alpha.unix.Chroot -analyzer-disable-checker alpha.unix.PthreadLock -analyzer-disable-checker alpha.unix.SimpleStream -analyzer-disable-checker alpha.unix.Stream -analyzer-disable-checker alpha.unix.cstring.BufferOverlap -analyzer-disable-checker alpha.unix.cstring.NotNullTerminated -analyzer-disable-checker alpha.unix.cstring.OutOfBounds -analyzer-checker=apiModeling.google.GTest -analyzer-checker=core.CallAndMessage -analyzer-checker=core.DivideZero -analyzer-checker=core.DynamicTypePropagation -analyzer-checker=core.NonNullParamChecker -analyzer-checker=core.NullDereference -analyzer-checker=core.StackAddressEscape -analyzer-checker=core.UndefinedBinaryOperatorResult -analyzer-checker=core.VLASize -analyzer-checker=core.builtin.BuiltinFunctions -analyzer-checker=core.builtin.NoReturnFunctions -analyzer-checker=core.uninitialized.ArraySubscript -analyzer-checker=core.uninitialized.Assign -analyzer-checker=core.uninitialized.Branch -analyzer-checker=core.uninitialized.CapturedBlockVariable -analyzer-checker=core.uninitialized.UndefReturn -analyzer-checker=cplusplus.NewDelete -analyzer-checker=cplusplus.NewDeleteLeaks -analyzer-checker=cplusplus.SelfAssignment -analyzer-checker=deadcode.DeadStores -analyzer-disable-checker debug.AnalysisOrder -analyzer-disable-checker debug.ConfigDumper -analyzer-disable-checker debug.DumpCFG -analyzer-disable-checker debug.DumpCallGraph -analyzer-disable-checker debug.DumpCalls -analyzer-disable-checker debug.DumpDominators -analyzer-disable-checker debug.DumpLiveVars -analyzer-disable-checker debug.DumpTraversal -analyzer-disable-checker debug.ExprInspection -analyzer-disable-checker debug.Stats -analyzer-disable-checker debug.TaintTest -analyzer-disable-checker debug.ViewCFG -analyzer-disable-checker debug.ViewCallGraph -analyzer-disable-checker debug.ViewExplodedGraph -analyzer-disable-checker ericsson.cpp.InvalidatedIterator -analyzer-checker=ericsson.cpp.MisuseEnumAsCondition -analyzer-checker=ericsson.cpp.PredWithState -analyzer-checker=ericsson.cpp.RaiiMisuse -analyzer-checker=ericsson.cpp.StrlenMisuseAlloc -analyzer-checker=ericsson.cpp.SuspiciousPointerArithmetic -analyzer-checker=ericsson.cpp.SuspiciousSizeof -analyzer-checker=ericsson.cpp.VirtualDtorPoly -analyzer-checker=ericsson.cpp.stl.ContainerOfAutoptr -analyzer-checker=ericsson.cpp.stl.MisusedRemove -analyzer-checker=ericsson.cpp.stl.PolymorphContainer -analyzer-checker=ericsson.cpp.stl.RevIterBase -analyzer-checker=ericsson.cpp.style.CatchByValue -analyzer-disable-checker ericsson.cpp.style.Goto -analyzer-checker=ericsson.cpp.style.LargeObjectPassed -analyzer-disable-checker ericsson.cpp.style.MagicNumbers -analyzer-disable-checker ericsson.cpp.style.MissingConst -analyzer-disable-checker ericsson.cpp.style.New -analyzer-disable-checker ericsson.cpp.style.NonPrivateData -analyzer-disable-checker ericsson.cpp.style.ReduceScope -analyzer-disable-checker ericsson.cpp.style.ReinterpretCast -analyzer-disable-checker ericsson.linuxkernelstyle.PointerDecl -analyzer-disable-checker ericsson.misrac.AssignmentInCondition -analyzer-disable-checker ericsson.misrac.ExternalArrayWithUnknownSize -analyzer-disable-checker ericsson.misrac.FunctionsWithNoParam -analyzer-disable-checker ericsson.misrac.IfFollowedByCompound -analyzer-disable-checker ericsson.misrac.SwitchDefaultBranch -analyzer-disable-checker ericsson.mtas.DbnDelayAfterRetry -analyzer-disable-checker ericsson.mtas.DialogueSetupCreate -analyzer-disable-checker ericsson.mtas.EpctParameters -analyzer-disable-checker ericsson.mtas.IllegalTracing -analyzer-disable-checker ericsson.mtas.LicenseManagement -analyzer-disable-checker ericsson.mtas.MtasConventions -analyzer-disable-checker ericsson.mtas.SerializeVersion -analyzer-disable-checker ericsson.mtas.SerializeWithoutObserver -analyzer-checker=ericsson.precpp11.stl.AllocWithState -analyzer-checker=ericsson.precpp11.stl.BinFunctorTypeMismatch -analyzer-checker=ericsson.precpp11.stl.UnaryFunctorTypeMismatch -analyzer-disable-checker ericsson.tsp.DbnTransactionManagement -analyzer-disable-checker ericsson.tsp.PointerCompare -analyzer-disable-checker ericsson.tsp.TspBuiltinTypes -analyzer-disable-checker ericsson.utility.WarningToReportConverter -analyzer-disable-checker llvm.Conventions -analyzer-checker=nullability.NullPassedToNonnull -analyzer-checker=nullability.NullReturnedFromNonnull -analyzer-disable-checker nullability.NullableDereferenced -analyzer-disable-checker nullability.NullablePassedToNonnull -analyzer-disable-checker nullability.NullableReturnedFromNonnull -analyzer-disable-checker optin.cplusplus.VirtualCall -analyzer-disable-checker optin.mpi.MPI-Checker -analyzer-disable-checker optin.osx.cocoa.localizability.EmptyLocalizationContextChecker -analyzer-disable-checker optin.osx.cocoa.localizability.NonLocalizedStringChecker -analyzer-disable-checker optin.performance.Padding -analyzer-disable-checker optin.portability.UnixAPI -analyzer-disable-checker osx.API -analyzer-disable-checker osx.NumberObjectConversion -analyzer-disable-checker osx.ObjCProperty -analyzer-disable-checker osx.SecKeychainAPI -analyzer-disable-checker osx.cocoa.AtSync -analyzer-disable-checker osx.cocoa.ClassRelease -analyzer-disable-checker osx.cocoa.Dealloc -analyzer-disable-checker osx.cocoa.IncompatibleMethodTypes -analyzer-disable-checker osx.cocoa.Loops -analyzer-disable-checker osx.cocoa.MissingSuperCall -analyzer-disable-checker osx.cocoa.NSAutoreleasePool -analyzer-disable-checker osx.cocoa.NSError -analyzer-disable-checker osx.cocoa.NilArg -analyzer-disable-checker osx.cocoa.NonNilReturnValue -analyzer-disable-checker osx.cocoa.ObjCGenerics -analyzer-disable-checker osx.cocoa.RetainCount -analyzer-disable-checker osx.cocoa.SelfInit -analyzer-disable-checker osx.cocoa.SuperDealloc -analyzer-disable-checker osx.cocoa.UnusedIvars -analyzer-disable-checker osx.cocoa.VariadicMethodTypes -analyzer-disable-checker osx.coreFoundation.CFError -analyzer-disable-checker osx.coreFoundation.CFNumber -analyzer-disable-checker osx.coreFoundation.CFRetainRelease -analyzer-disable-checker osx.coreFoundation.containers.OutOfBounds -analyzer-disable-checker osx.coreFoundation.containers.PointerSizedValues -analyzer-checker=security.FloatLoopCounter -analyzer-checker=security.insecureAPI.UncheckedReturn -analyzer-checker=security.insecureAPI.getpw -analyzer-checker=security.insecureAPI.gets -analyzer-checker=security.insecureAPI.mkstemp -analyzer-checker=security.insecureAPI.mktemp -analyzer-checker=security.insecureAPI.rand -analyzer-disable-checker security.insecureAPI.strcpy -analyzer-checker=security.insecureAPI.vfork -analyzer-checker=unix.API -analyzer-checker=unix.Malloc -analyzer-checker=unix.MallocSizeof -analyzer-checker=unix.MismatchedDeallocator -analyzer-checker=unix.StdCLibraryFunctions -analyzer-checker=unix.Vfork -analyzer-checker=unix.cstring.BadSizeArg -analyzer-checker=unix.cstring.NullArg -analyzer-checker=valist.CopyToSelf -analyzer-checker=valist.Uninitialized -analyzer-checker=valist.Unterminated -analyzer-config xtu-dir=./report_debug/ctu-dir/i386 -analyzer-display-ctu-progress -o ./sources-root/local/scratch/repos/ML66/reports/bns__2017_11_16__15_34_00/Alarm.cc_7817343c04f25df6faf81ba12c6298da.plist -x c++ ./sources-root/local/scratch/repos/ML66/ml66_nps/nps/bns_app_src/platform/eqm/lich/src/Alarm.cc
1.      <eof> parser at end of file
2.      While analyzing stack:
        #0 void insert_restart_required_message()
3.      ./sources-root/local/scratch/repos/ML66/ml66_nps/nps/bns_app_src/platform/eqm/lich/src/Alarm.cc:743:35: Error evaluating statement
4.      ./sources-root/local/scratch/repos/ML66/ml66_nps/nps/bns_app_src/platform/eqm/lich/src/Alarm.cc:743:35: Error evaluating statement
clang-5.0: error: unable to execute command: Aborted (core dumped)
clang-5.0: error: clang frontend command failed due to signal (use -v to see invocation)
clang version 5.0.0 (git@github.com:martong/clang.git cfdfecd3398c2645b04a1b44d25636ee82cae56a) (git@github.com:llvm-mirror/llvm.git 657c31173ea30090583e40c7a9204561d9c2d8c4)
Target: powerpc-montavista-linux-gnu
Thread model: posix
InstalledDir: /home/egbomrt/WORK/llvm/build/debug/bin
clang-5.0: note: diagnostic msg: PLEASE submit a bug report to http://llvm.org/bugs/ and include the crash backtrace, preprocessed source, and associated run script.
```

--- a/lib/AST/DeclBase.cpp
+++ b/lib/AST/DeclBase.cpp
@@ -1321,6 +1321,8 @@ bool DeclContext::containsDecl(Decl *D) const {
           (D->NextInContextAndBits.getPointer() || D == LastDecl));
 }
 
+static bool shouldBeHidden(NamedDecl *D);
+
 void DeclContext::removeDecl(Decl *D) {
   assert(D->getLexicalDeclContext() == this &&
          "decl being removed from non-lexical context");
@@ -1350,6 +1352,10 @@ void DeclContext::removeDecl(Decl *D) {
   // Remove D from the lookup table if necessary.
   if (isa<NamedDecl>(D)) {
     NamedDecl *ND = cast<NamedDecl>(D);
+
+    // Do not try to remove the declaration if that is invisible to qualified
+    // lookup.  E.g. template sepcializations are skipped.
+    if (shouldBeHidden(ND)) return;
 
     // Remove only decls that have a name
     if (!ND->getDeclName()) return;

--- a/unittests/AST/ASTImporterTest.cpp
+++ b/unittests/AST/ASTImporterTest.cpp
@@ -554,6 +554,38 @@ TEST(ImportDecl, ImportFunctionTemplateDecl) {
               has(returnStmt(has(integerLiteral(equals(2)))))))))))))))))));
 }
 
+TEST(ImportExpr, ImportClassTemplatePartialSpecialization) {
+  MatchVerifier<Decl> Verifier;
+  EXPECT_TRUE(testImport(
+      R"s(
+// excerpt from <functional>
+
+namespace declToImport {
+
+template <typename _MemberPointer>
+class _Mem_fn;
+
+template <typename _Tp, typename _Class>
+_Mem_fn<_Tp _Class::*> mem_fn(_Tp _Class::*);
+
+template <typename _Res, typename _Class>
+class _Mem_fn<_Res _Class::*> {
+    template <typename _Signature>
+    struct result;
+
+    template <typename _CVMem, typename _Tp>
+    struct result<_CVMem(_Tp)> {};
+
+    template <typename _CVMem, typename _Tp>
+    struct result<_CVMem(_Tp&)> {};
+};
+
+} // namespace
+          )s",
+      Lang_CXX, "",
+      Lang_CXX, Verifier, namespaceDecl()));
+}
+
 TEST(ImportExpr, ImportUnresolvedLookupExpr) {
 MatchVerifier<Decl> Verifier;
 EXPECT_TRUE(

